code/controllers/subsystem/bluespace_exploration.dm

```diff
@@ -0,0 +1,391 @@
+#define CLEAR_TURF_PROCESSING_TIME 600
+
+SUBSYSTEM_DEF(bluespace_exploration)
+	name = "Bluespace Exploration"
+	wait = 1
+	priority = FIRE_PRIORITY_EXPLORATION
+	init_order = INIT_ORDER_BS_EXPLORATION
+
+	var/generating = FALSE
+
+	//Starmap generation
+	var/datum/star_system/current_system = null
+	var/list/star_systems = list()
```

Comment:
Reminder that lists should be initialized either in Initialize() or preferably not until they're actually needed.

Reply:
I mean, sure but it is a subsystem so only done once making it a near unone noticeable performance boost